source/common/upstream/logical_dns_cluster.cc

```diff
@@ -21,8 +21,7 @@ LogicalDnsCluster::LogicalDnsCluster(const Json::Object& config, Runtime::Loader
   }
 
   dns_url_ = hosts_json[0]->getString("url");
-  Network::Utility::hostFromTcpUrl(dns_url_);
-  Network::Utility::portFromTcpUrl(dns_url_);
+  hostname_ = Network::Utility::hostAndPortFromTcpUrl(dns_url_);
```

Comment:
@kyessenov is this feature actually required for Istio? @rshriram I thought you were adding this to deal with cloud foundry.